sdk/cosmosdb/cosmos/src/globalEndpointManager.ts

```diff
@@ -183,6 +187,45 @@ export class GlobalEndpointManager {
     }
   }
 
+  private refreshStaleUnavailableLocations(): void {
+    const now = Date.now();
+    this.updateLocation(now, this.unavailableReadableLocations, this.readableLocations);
+    this.unavailableReadableLocations = this.cleanUnavailableLocationList(
+      now,
+      this.unavailableReadableLocations
+    );
+
+    this.updateLocation(now, this.unavailableWriteableLocations, this.writeableLocations);
+    this.unavailableWriteableLocations = this.cleanUnavailableLocationList(
+      now,
+      this.unavailableWriteableLocations
+    );
+  }
+  private updateLocation(now: number, unavailableLocations: Location[], allLocations: Location[]) {
+    for (const location of unavailableLocations) {
+      const unavaialableLocation = allLocations.find((loc) => loc.name === location.name);
+      if (
+        unavaialableLocation.locationUnavailability &&
```

Comment:
we are searching in a list. `unavaialableLocation` could theoretically be undefined. (May be due to change in server-side behavior). Please check for this possibility as well. i.e., 
`if (unavaialableLocation && unavaialableLocation.locationUnavailability) ...)`

Reply:
addressed